File: apps/api/src/routes/prompts.ts
import { Elysia, t } from 'elysia';
import { db } from '../store';
import { Prompt } from '../models';

export const promptsRoutes = new Elysia({ prefix: '/prompts' })
  .get('/', () => db.getPrompts(), {
    response: t.Array(Prompt)
  });

File: crates/relaycode-api/src/routes/prompts.rs
use relaycode_schema::Prompt;
use relaycode_core::STORE;
use axum::{routing::get, Json, Router};

#[utoipa::path(
    get,
    path = "/api/prompts",
    tag = "Prompts",
    responses(
        (status = 200, body = [Prompt])
    )
)]
pub async fn list_prompts() -> Json<Vec<Prompt>> {
    Json(STORE.get_prompts())
}

pub fn router() -> Router {
    Router::new()
        .route("/prompts", get(list_prompts))
}
